### PR TITLE
Cr/29952 Misleading Coverage

### PR DESCRIPTION
**Pull request by stahlbauer**
Addresses issue [TS-29952](https://cqse.atlassian.net/browse/TS-29952)

- [x] Changes are tested adequately
- [x] CHANGELOG.md of the affected packages are updated

Please respect the vote of the [Teamscale bot](https://demo.teamscale.com) or flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config needs adjustment, please state so in a comment and discuss this with your reviewer.

### CODE REVIEW

File: packages/teamscale-coverage-collector/src/receiver/Session.ts [hunk 1]
```diff
@@ -64,12 +64,39 @@ export class Session {
...
+			// Determine the column range to consider for this line
+			let scanFromColumn;
+			if (line === startLine) {
+				scanFromColumn = startColumn;
+			} else {
+				scanFromColumn = 0;
+			}
+			// Since we do not know the length of the different lines, we assume
+			// all to end in `endColumn`.
+			const scanToColumn = endColumn;
```
Comment (DreierF) [inline packages/teamscale-coverage-collector/src/receiver/Session.ts:79]: Will this produce incorrect results for this cause issues for `startLine=1, startColumn=20, endLine=2, endColumn=10`?

Then we have `scanFromColumn=20` and `scanToColumn=10`

File: packages/teamscale-coverage-collector/src/receiver/Session.ts [hunk 2]
```diff
@@ -64,12 +64,39 @@ export class Session {
...
+					if (lastCoveredLine !== originalPosition.line) {
+						this.storage.putCoverage(this.projectId, originalPosition.source, [originalPosition.line]);
+						lastCoveredLine = originalPosition.line;
+					}
+				} else {
+					this.storage.signalUnmappedCoverage(this.projectId);
+				}
+
+				// Step to the next column to map back to the original
+				column = column + Math.max(1, (originalPosition.name?.length ?? 1));
```
Comment (DreierF) [inline packages/teamscale-coverage-collector/src/receiver/Session.ts:95]: What is `originalPosition.name` and why can we derive the next column position from it? Maybe an additional inline comment could help explain things

**Review by DreierF — CHANGES_REQUESTED** [review]

File: packages/teamscale-coverage-collector/src/receiver/Session.ts (hunk 1 above)
Comment (stahlbauer) [reply packages/teamscale-coverage-collector/src/receiver/Session.ts:79]: Good catch. Fixed this and added a ticket for a better line length estimation.

File: packages/teamscale-coverage-collector/src/receiver/Session.ts (hunk 2 above)
Comment (stahlbauer) [reply packages/teamscale-coverage-collector/src/receiver/Session.ts:95]: I have added a comment on this.

**Review by DreierF — APPROVED** [review]

**Pull request merged**